pynecone/app.py

```diff
@@ -234,6 +234,38 @@ def _check_routes_conflict(self, new_route: str):
                         f"You cannot use multiple catchall for the same dynamic path ({route} !== {new_route})"
                     )
 
+    def add_custom_404_page(self, component, title=None, image=None, description=None):
```

Comment:
What if we make it so the user never has to call this method, instead when they do
```
app.add_page(my_404_component, path="/404")
```
we can call this if we detect the path is `/404`? Just to reduce the number of methods we expose to the end user.

Then also a lot of the common logic for the `add_meta`, etc can stay in one function and this one only handles modifying the route.

Reply:
Talked offline - let's keep this since it provides clarity/flexibility. We can possibly combine it into `add_page` with a flag later.